templates/tensorflow-gpu-1.12/image_classifier/src/helpers/download_helpers.py

```diff
@@ -0,0 +1,100 @@
+import pickle
+import urllib
+import os
+from Algorithmia.errors import AlgorithmException
+from tempfile import mkdtemp
+
+def download_image(client, image_url, size=None):
+	"""
+	Downloads the image file from the provided URL and returns the path to the locally saved file.
+	NOTE: This method uses the util/SmartImageDownloader algorithm to perform the task and this implies
+	that your algorithm should be able to call other algorithms and must have internet access.
+	Parameters:
+	client: Algorithmia python client
+	image_url: Web url to the image that needs to be fetched
+	dimensions: If provided, smart image downloader will attempt to format the image to fit the dimensions as a square, like 224x224
+	"""
+	if size:
+		algo_input = {
+			'image': image_url,
+			'resize': {
+				'width': size,
+				'height': size
+			},
+		}
+	else:
+		algo_input = image_url
+	img_data_path = client.algo("util/SmartImageDownloader").pipe(algo_input).result["savePath"][0]
+	return client.file(img_data_path).getFile().name
+
+def load_pickle_file(pickle_file_path, local_file = True, client=None):
+	"""
+	Download and load a pickle file into a python object. The file can exist in Algorithmia (data://) or any public url or locally
+	Parameters:
+	location: Absolute path to the file (remote or local)
```

Comment:
not going to judge any of the actual python code much but i think there's conventions out there for specifying parameter names in docstrings that numpy uses that other tools work well with.  something like `:param location str: blah blah` https://stackoverflow.com/a/40596167

Reply:
I like doc strings but they aren't python 2 compatible normally (whcih we still support blah)